rust/crd/src/lib.rs

```diff
@@ -401,12 +420,23 @@ impl DbType {
     }
 }
 
+/// Database connection specification for the metadata database.
 #[derive(Clone, Debug, Default, Deserialize, Eq, Hash, JsonSchema, PartialEq, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct DatabaseConnectionSpec {
+    /// A connection string for the database. For example:
+    /// `jdbc:postgresql://hivehdfs-postgresql:5432/hivehdfs`
     pub conn_string: String,
+
+    /// The database user.
     pub user: String,
+
+    /// The password for the database user.
     pub password: String,
+
+    /// The type of database to connect to. Supported are:
+    /// `postgres`, `mysql`, `oracle`, `mssql` and `derby`.
```

Comment:
This supported list would show up in the docs of the enum? If yes i would not explicitly state this here (or at least a full list because it may be prone to changes)

Reply:
The docs generator only shows "type string" and not the enum values unfortunately. That's why I listed the values here, I think it's important to list them.

I understand your reasoning though. But do you think it would be fine to include anyways? I don't think the list will realistically change anymore.

Reply:
No then its fine with me. Could we add docs to the single enum fields? Like the Postgres (and link), derby (link and warning that this is not for production / HA)?